From: Emmanuel Arias <eamanu@yaerobi.com>
Date: Fri, 6 Jan 2023 01:18:41 -0300
Subject: Stop using rapidfuzz. Revert upstream change in Debian.

Upstream introduce rapidfuzz instead of pylev because the first one
improve performance on find_similar_names. This brekas the build in
Debian because rapidfuzz is not in Debian yet. This patch revert that
change. We should remove this patch when rapidfuzz is in Debian. Also,
add pylev in pyproject.toml file.
---
 src/cleo/_utils.py | 11 ++++++++---
 1 file changed, 8 insertions(+), 3 deletions(-)

Index: python-cleo/src/cleo/_utils.py
===================================================================
--- python-cleo.orig/src/cleo/_utils.py
+++ python-cleo/src/cleo/_utils.py
@@ -5,7 +5,7 @@ import math
 from dataclasses import dataclass
 from html.parser import HTMLParser
 
-from rapidfuzz.distance import Levenshtein
+from pylev import levenshtein
 
 
 class TagStripper(HTMLParser):
@@ -53,10 +53,11 @@ def find_similar_names(name: str, names:
     """
     threshold = 1e3
     distance_by_name = {}
+    suggested_names = []
 
     for actual_name in names:
         # Get Levenshtein distance between the input and each command name
-        distance = Levenshtein.distance(name, actual_name)
+        distance = levenshtein(name, actual_name)
 
         is_similar = distance <= len(name) / 3
         substring_index = actual_name.find(name)
@@ -75,7 +76,10 @@ def find_similar_names(name: str, names:
         if value[0] < 2 * threshold
     }
     # Display results with shortest distance first
-    return sorted(distance_by_name, key=lambda key: distance_by_name[key])
+    for k, _v in sorted(distance_by_name.items(), key=lambda i: (i[1][0], i[1][1])):
+        if k not in suggested_names:
+            suggested_names.append(k)
+    return suggested_names
 
 
 @dataclass
Index: python-cleo/pyproject.toml
===================================================================
--- python-cleo.orig/pyproject.toml
+++ python-cleo/pyproject.toml
@@ -36,6 +36,7 @@ classifiers = [
 python = "^3.7"
 crashtest = "^0.4.1"
 rapidfuzz = "^3.0.0"
+pylev = "^1.4.0"
 
 [tool.poetry.group.dev.dependencies]
 mypy = [
